easybuild/tools/run.py

```diff
@@ -80,16 +80,18 @@ def run_cmd_cache(func):
     @functools.wraps(func)
     def cache_aware_func(cmd, *args, **kwargs):
         """Retrieve cached result of selected commands, or run specified and collect & cache result."""
-
-        # fetch from cache if available, cache it if it's not
-        if cmd in cache:
-            _log.debug("Using cached value for command '%s': %s", cmd, cache[cmd])
-            return cache[cmd]
+        # fetch from cache if available, cache it if it's not, but only on cmd strings
```

Comment:
Since we're only using strings in `CACHED_COMMANDS`, why do we need to check the type of `cmd` at all?

Since we never cache a non-string `cmd` right now, `cmd in cache` will always be `False`, and so we'll always call out to the actual `func`?

This looks like a case of premature optimisation here, especially for a command that's executed very rarely...